src/nvim/syntax.c

```diff
@@ -7186,20 +7217,27 @@ int syn_namen2id(char_u *linep, int len)
 int syn_check_group(char_u *pp, int len)
 {
   char_u  *name = vim_strnsave(pp, len);
+
+  // HACK we need to make sure that the first highlight is the "Normal" hl
+  if (highlight_ga.ga_len == 0) {
+    int res = syn_add_group((char_u *)xstrdup("Normal"));
+    assert(res == 0);
```

Comment:
Indeed. I would like to move the Normal hl creation to init_highlight but that would need more thoughts/tests as I am afraid ` :high Normal NONE`  or some other command could break things.